provider/kubernetes/kubernetes.go

```diff
@@ -247,8 +247,14 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 					templateObjects.Backends[r.Host+pa.Path].LoadBalancer.Method = "drr"
 				}
 
-				if service.Annotations[types.LabelBackendLoadbalancerSticky] == "true" {
-					templateObjects.Backends[r.Host+pa.Path].LoadBalancer.Sticky = true
+				if len(service.Annotations[types.LabelBackendLoadbalancerSticky]) != 0 {
+					log.Warn("Deprecated configuration found: %s. Please use %s.", types.LabelBackendLoadbalancerSticky, types.LabelBackendLoadbalancerStickiness)
+				}
+
+				if service.Annotations[types.LabelBackendLoadbalancerSticky] == "true" || service.Annotations[types.LabelBackendLoadbalancerStickiness] == "true" {
+					templateObjects.Backends[r.Host+pa.Path].LoadBalancer.Stickiness = &types.Stickiness{
+						CookieName: cookie.GenerateName(r.Host + pa.Path),
```

Comment:
Why not choosing the cookie name generation from hash here ? :thinking:

Reply:
This is a case: `GenerateName` generate a hash as name

Reply:
Oops, got it 😊